components/modelregistry/modelregistry.go

```diff
@@ -124,3 +140,25 @@ func (m *ModelRegistry) ReconcileComponent(ctx context.Context, cli client.Clien
 	}
 	return nil
 }
+
+func (m *ModelRegistry) createDependencies(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	// create DefaultModelRegistryCert
+	if err := cluster.PropagateDefaultIngressCertificate(ctx, cli, DefaultModelRegistryCert, dscispec.ServiceMesh.ControlPlane.Namespace); err != nil {
```

Comment:
is the whole solution based on servicemesh must be installed and enabled?

Reply:
Yes, servicemesh is a pre-requisite for model registry. But, I thought servicemesh feature was going to be enabled by default, is that not the case anymore?

Reply:
It is enabled by default , but can be set to removed if Kserve is disabled. I think we would need to add checks for Model registry to return error when servicemesh is not enabled

Reply:
maybe not related to this PR, but why we do not set owner on these secret? if isSecretOutdated() is only called when component reconcile happens, then there is possibility, default ingress cert gets updated, but our secret is out of date, value in the params.env is out of data too, or?

Reply:
There is a [watch](https://github.com/opendatahub-io/opendatahub-operator/blob/eee87c016f1f23f82f02ee00113f5bd0345a03fe/controllers/datasciencecluster/datasciencecluster_controller.go#L462) on the source cert, and as far as I can tell it triggers a reconcile of dsc.

Reply:
The value in params.env is just the secret name, so that doesn't change anything in the MR operator config. 

And it doesn't really need to, since the name is actually passed to the Istio ingress gateway, which should be watching that secret itself and reload it when it changes.

Reply:
thanks, i forgot the "watch" from our operator, which is used when original ingress cert get expired case.